docs/modules/ROOT/pages/troubleshooting/index.adoc

```diff
@@ -44,4 +44,110 @@ If you see this error message, it means that the pipeline run has exceeded the d
 spec:
   # ...
   timeout: 1h30m0s  # increase accordingly
-----
\ No newline at end of file
+----
+
+== Failed to push or pull image
+
+Failing to authenticate with a container registry can be hard to debug. To be able to troubleshoot
+effectively, one needs to understand how registry authentication works in Tekton. For the full
+details, see the Tekton link:https://tekton.dev/docs/pipelines/auth/[Authentication] docs. In short:
+
+. Tekton uses a ServiceAccount (named `appstudio-pipeline` by default in Konflux) to run your Pipeline.
+. This ServiceAccount has a list of `imagePullSecrets` and `secrets`.
+.. Tekton uses the `imagePullSecrets` when pulling the images that execute the Tasks in your pipeline.
+   These images are typically hardcoded in the Task definition and publicly accessible. As a Konflux
+   user, you're more interested in the `secrets`.
```

Comment:
Are they more interested in this case or generally?

Reply:
Almost always. The only time `imagePullSecrets` are relevant is if they're developing their own Task and using a private image as the `image:` to run a step

Or if Konflux devs screw up and use a private image for a Konflux task